### PR TITLE
make addBuilds support attach different product_version images 

### PR DESCRIPTION
**Pull request by shiywang**
…e same time

### CODE REVIEW

File: errata_tool/erratum.py [hunk 1]
```diff
@@ -616,65 +616,95 @@ def addCC(self, email):
...
-            if type(f) is not str:
+            if not isinstance(f, str):
                 raise ValueError('flag ' + str(f) + ' is not a string')
         for f in flags:
             if f in self.current_flags:
                 self.current_flags.remove(f)
 
     # Adding and removing builds can't be done atomically.  Wondering whether
-    def addBuildsDirect(self, buildlist, release, **kwargs):
+    def addBuildsDirect(self, buildlist, default_product_version, **kwargs):
```
Comment (ktdreyer) [inline errata_tool/erratum.py]: I like this idea although this changes the API signature for this method.

@yazug @lhh can you please confirm that you're not using `addBuildsDirect()` in your code? (only `addBuilds()`)?

Comment (shiywang) [reply errata_tool/erratum.py]: oh, about the name I can move it back to `release` although I don't like the term `release`.
I am trying to do none breaking change about `addBuildsDirect` and `addBuilds` base on the type of `buildlist` and value of `release`.
so @yazug @lhh let me know if something isn't appropriate.

Comment (yazug) [reply errata_tool/erratum.py]: We do use addBuildsDirect by position, not by key-value parameters

File: errata_tool/erratum.py [hunk 2]
```diff
@@ -557,7 +557,7 @@ def setState(self, state):
                                   state.upper())
 
     def _addBug(self, b):
-        if type(b) is not int:
+        if not isinstance(b, int):
```
Comment (yazug) [inline errata_tool/erratum.py]: This linting cleanup could be split out and merged by inspection, as it is not directly related to this pull request.

File: errata_tool/erratum.py (hunk 1 above)
Comment (yazug) [reply errata_tool/erratum.py]: we are using addBuildsDirect() but only by position parameters when we are looping through adding builds to the advisory

Comment (yazug) [reply errata_tool/erratum.py]: Also we should add doctext to addBuildsDirect for how things are being passed in and make sure to add unittests for existing and anticipated calling modes.

File: errata_tool/erratum.py [hunk 3]
```diff
@@ -722,7 +754,7 @@ def setFileInfo(self, file_info):
         self._processResponse(r)
 
     def removeBuilds(self, buildlist):
-        if type(buildlist) is not str and type(buildlist) is not list:
+        if not isinstance(buildlist, str) and not isinstance(buildlist, list):
```
Comment (yazug) [inline errata_tool/erratum.py]: different issue but str here should probably be six.string_types.  Will open separate issue for this

Comment (yazug) [reply errata_tool/erratum.py]: https://github.com/red-hat-storage/errata-tool/issues/181

File: errata_tool/erratum.py (hunk 2 above)
Comment (yazug) [reply errata_tool/erratum.py]: not a problem here

File: errata_tool/erratum.py (hunk 3 above)
Comment (yazug) [reply errata_tool/erratum.py]: sorry for noise not a problem, verified it was correct behavior.

**Comment by shiywang** [comment]
@yazug @ktdreyer thank you for the review close this cuz we decided to take another approach which doesn't need to touch code in here.

**Pull request closed without merging**